### DISCUSSION TITLE
Multiple Instances Same Raspberry Pi

### DISCUSSION THREAD

**Discussion by ghost** [awawa-dev/HyperHDR#799] [Q&A]
Hi. I have my tv backlights all setup as my first instance using my pi and PWM GPIO 18. It works amazing. I was hoping to connect 2  more instances. I built a copycat phillips hue floor gradient lamp. I have it all wired to connect into PWM GPIO pin 12 and 13 on the same pi as my first instance. When I connected my first floor lamp and installed the instance in HyperHDR everything went haywire. Sometimes my original tv lights would flicker other times they would just freeze at what ever color they are and then not change. When I delete the 2nd instance and restart the pi everything works again. Is there a limitation on instances to 1 pi? If so I was going to ESP32 and WLED as a driver for my 2 floor lamps. If there are better ways I would love to hear them!  Thanks for any help!

**Comment by awawa-dev** [comment]
Hi
Maybe you need to change DMA channel for other instances as well? Since this is an unofficially supported configuration on RPi, I can't be of much help here. Maybe you will find something on the ws281x project homesite.

**Reply by ghost** [reply]
I will give that a try, thanks for the response. Curious I noticed when i was installing this was not officially supported. I followed videos online to download. Is it the pi that is not supported or the light strip? Is there a better way for me to set it all up so it is officially supported?

**Reply by awawa-dev** [reply]
PWM method on Rpi: https://github.com/awawa-dev/HyperHDR/wiki/FAQ beside breaking changes in RPi 5 hardware design caused that the method doesn't work at all on that board. Using eg. HyperSerialPico allows to handle up to two similar length LED strip at once (multi-segment) with one instance.

**Reply by ghost** [reply]
that is really helpful thanks! I am going to go the ESP32 route to act as the driver. I already have level shifters wired in and everything. I originally tried using HyperserialSPI but was unable to download the firmware. When I created the fork and followed all the prompts no actions were triggered. I may give that another try. Is there any way to do a network connection between the pi and esp32? would that be where wled comes in?

**Reply by awawa-dev** [reply]
Did you enable Github Action **before** making changes to the configuration? Because changing the files triggers the build (not enabling Github Action alone) and if the Github Action was disabled at the moment it wont build it later when you enable it finally. Make some new changes even to readme.md and save it. It should build it now.

**Reply by ghost** [reply]
I will double check, i saw that part in the instructions. I will give it a try right now and report back. Hopefully can get it downloaded!

**Reply by ghost** [reply]
I was doing something wrong, It looks like it is working now. I will use this route for those 2 new instances. Thanks for the help!

**Reply by ghost** [reply]
This could be a dumb question but will I be able to connect 2 ESP32's running HyperSPI to my Pi 4? If so how do I go about changing the connecting pin configuration for the second ESP32 to PI?

**Reply by awawa-dev** [reply]
I haven't tried it, but it's theoretically possible since RPI has two SPI interfaces (SPI0 and SPI1). First, of course, you need to enable SPI1 in the system. I don't know if SPI1 achieves the same high speeds as SPI0, from what I remember they are slightly different. You can always connect one ESP via HyperSPI and the other via HyperSerialESP32. Or two ESPs via HyperSerialESP32 because COMs have no limit.

**Reply by ghost** [reply]
That makes sense, ok thank you.

**Comment by ghost** [comment]
This would be correct right? this is what I had it set to. I am sure it is user error somewhere in the process. 

![Screenshot 2024-03-21 at 12 12 55 PM](https://github.com/awawa-dev/HyperHDR/assets/124069573/028a3c02-52e1-4f44-905f-75aea02d0c15)

**Reply by awawa-dev** [reply]
I can see on your fork that already enabled Github Action. Now make some changes to readme.md (eg add a space, it doesnt matter where in that file) and save changes.

**Reply by ghost** [reply]
I was doing something wrong, It looks like it is working now. I will use this route for those 2 new instances. Thanks for the help!
